### PR TITLE
Allow for non-object resources in authorize

### PR DESCRIPTION
**Pull request by Zuluru**
Addresses issue #60.

### CODE REVIEW

File: tests/test_app/TestApp/Policy/StringResolver.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+/**
+ * Very simple policy resolver that accepts string policy names.
+ */
+class StringResolver implements ResolverInterface
+{
+    /**
+     * Get a policy for a string.
+     *
+     * @param string $resource The resource.
+     * @return object
```
Comment (dereuromark) [inline tests/test_app/TestApp/Policy/StringResolver.php]: According to used class (constructor of it), this is `\Authorization\Policy\ResolverInterface`

Comment (Zuluru) [reply tests/test_app/TestApp/Policy/StringResolver.php]: The @return doc? That was copied directly from the OrmResolver.

Comment (Zuluru) [reply tests/test_app/TestApp/Policy/StringResolver.php]: Being stricter seems like a good idea, but is outside the scope of this PR, I think. Addresses a different issue.

File: src/Controller/Component/AuthorizationComponent.php [hunk 2]
```diff
@@ -69,7 +69,7 @@ public function authorize($resource, $action = null)
             return;
         }
 
-        throw new ForbiddenException([$action, get_class($resource)]);
+        throw new ForbiddenException([$action, is_object($resource) ? get_class($resource) : (is_string($resource) ? $resource : gettype($resource))]);
```
Comment (markstory) [inline src/Controller/Component/AuthorizationComponent.php]: I would avoid nested ternary operators in PHP they are a bit funny.

File: tests/test_app/TestApp/Policy/StringResolver.php (hunk 1 above)
Comment (markstory) [reply tests/test_app/TestApp/Policy/StringResolver.php]: I don't think Policies have a common interface though.

Comment (Zuluru) [reply tests/test_app/TestApp/Policy/StringResolver.php]: I think he was suggesting that they should? Which, like I said, would be a separate issue.

Comment (markstory) [reply tests/test_app/TestApp/Policy/StringResolver.php]: Yes it is a separate conversation.

File: src/Controller/Component/AuthorizationComponent.php [hunk 3]
```diff
@@ -69,7 +69,14 @@ public function authorize($resource, $action = null)
             return;
         }
 
-        throw new ForbiddenException([$action, get_class($resource)]);
+        if (is_object($resource)) {
+            $name = get_class($resource);
+        } else if (is_string($resource)) {
```
Comment (dereuromark) [inline src/Controller/Component/AuthorizationComponent.php]: elseif

**Comment by markstory** [comment]
Thank you 👍

**Pull request merged**